chef/cookbooks/cpe_utils/libraries/node_functions.rb

```diff
@@ -284,6 +284,30 @@ def virtual?
       end
     end
 
+    # Granular virtual device type for macOS
+    def virtual_macos_type
+      unless macos?
+        Chef::Log.warn('node.virtual_macos called on non-OS X!')
+        return
+      end
+      return node['virtual_macos'] if node['virtual_macos']
+      if node['hardware']['boot_rom_version'].include? 'VMW'
+        virtual_type = 'vmware'
+      elsif node['hardware']['boot_rom_version'].include? 'VirtualBox'
+        virtual_type = 'virtualbox'
+      else
+        virtual_type = Mixlib::ShellOut.new(
+          '/usr/sbin/system_profiler SPEthernetDataType',
+        ).run_command.stdout.to_s[/Vendor ID: (.*)/, 1]
+        if virtual_type.include? '0x1ab8'
+          virtual_type = 'parallels'
+        else
+          virtual_type = 'physical'
```

Comment:
If it is a physical machine, virtual_macos_type should probably return nil.

Reply:
maybe, but then it wouldn't match the standard this is supposed to copy (from munki-facts and other python tools). I'd prefer to keep it returning physical, but I won't be too upset if it's nil.

Reply:
returning a string is truthy. but this is fine. its not like he is defineing mac_vm? which the caller would expect a bool

Reply:
I wasn't thinking in terms of truthy. I was thinking this is asking "what type of virtual machine is this." If its physical, it is not a virtual machine at all, therefore `nil`. In other words, physical isn't a type of virtual machine.